src/renderer/ThemeBackend.js

```diff
@@ -0,0 +1,225 @@
+    messageAttachmentIconBg: '#ffffff', // Only changable with theme.raw
+    messageAttachmentFileName: '#070c14',
+    messageAttachmentFileSize: '#070c14',
+    // Login Screen
+    loginInputFocusColor: '#42A5F5',
+    loginButtonText: '#42A5F5',
+    deltaChatPrimaryFg: theme.textPrimary, // todo rename this var
+    deltaChatPrimaryFgLight: theme.textSecondary, // todo rename this var
+    // Context Menu
+    contextMenuBg: theme.bgSecondary,
+    contextMenuBorder: undefinedGuard(
+      theme.bgSecondary, c => changeContrast(c, 0.1)
+    ),
+    contextMenuText: theme.textSecondary,
+    contextMenuSelected: theme.bgSecondary,
+    contextMenuSelectedBg: '#a4a6a9',
+    // Misc
+    avatarLabelColor: '#ffffff', // Only changable with theme.raw
+    bp3DialogHeaderBg: theme.bgSecondary,
+    bp3DialogBg: theme.bgPrimary,
+    bp3DialogCardBg: theme.bgSecondary,
+    bp3Heading: theme.textPrimary,
+    bp3ButtonText: theme.textPrimary,
+    bp3ButtonBg: theme.bgPrimary,
+    brokenMediaText: '#070c14',
+    brokenMediaBg: '#ffffff',
+    unreadCountBg: theme.accentColor,
+    unreadCountLabel: '#ffffff', // Only changable with theme.raw
+    contactListItemBg: '#62656a',
+    errorColor: '#f44336',
+    globalLinkColor: '#2090ea', // Only changable with theme.raw
+    globalBackground: theme.bgPrimary,
+    globalText: theme.textPrimary,
+    mapOverlayBg: theme.bgPrimary,
+    videoPlayBtnIcon: theme.accentColor,
+    videoPlayBtnBg: '#ffffff', // Only changable with theme.raw
+    scrollBarThumb: undefinedGuard(
+      theme.scrollbarTransparency, c => Color('black').alpha(c).rgb().string()
+    ),
+    scrollBarThumbHover: undefinedGuard(
+      theme.scrollbarTransparency, c => Color('black').alpha(c + 0.14).rgb().string()
+    )
+
+  }
+  Object.keys(themeData).forEach(key => themeData[key] === undefined ? delete themeData[key] : '')
+  if (theme.raw) {
+    themeData = Object.assign(themeData, theme.raw)
+    console.log('theme.raw', theme.raw, themeData)
+  }
+  return themeData
+}
+
+export const defaultTheme = Object.freeze({
+  bgImagePath: '../images/background_hd2.svg',
+  bgPrimary: '#fff',
+  bgSecondary: '#f5f5f5',
+  accentColor: '#2090ea',
+  textPrimary: '#010101',
+  textSecondary: '#62656a',
+  ovalButtonBg: '#415e6b',
+  ovalButtonText: '#fff',
+  bgChatView: 'lime',
+  bgNavBar: '#415e6b',
+  textNavBar: '#fff',
+  scrollbarTransparency: 0.4
+
+})
+
+export const defaultThemeData = Object.freeze(ThemeDataBuilder(defaultTheme))
+
+export const ThemeVarOverwrite = (theme) => `
+--clr-avatar-label: ${theme.avatarLabelColor};
+--clr-bp3-dialog-bg: ${theme.bp3DialogBg};
+--clr-bp3-dialog-card-bg: ${theme.bp3DialogCardBg};
+--clr-bp3-heading: ${theme.bp3Heading}
+--clr-bp3-button-text: ${theme.bp3ButtonText};
+--clr-bp3-button-bg: ${theme.bp3ButtonBg}
+--clr-bp3-dialog-header-bg: ${theme.bp3DialogHeaderBg}
+--clr-broken-media-bg: ${theme.brokenMediaText};
+--clr-broken-media-bg-text: ${theme.brokenMediaBg};
```

Comment:
Even more hardcodes with different naming conventions and names, that's confusing. Can we streamline this? For example start using camelCase everywhere?

Reply:
This is setting the css variables and as far as i know the naming for those is --[name in lowercase + '-'] 
https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties

Reply:
I would propose following ideas:
1. use snake case name everywhere
2. use snake case for css custom properties, automatically (with help of a method) translate it into a camelCase name
3. use camelCase everywhere as it seems to work: https://codepen.io/anon/pen/xvXPWN?&editable=true

Reply:
you only need to "hardcode"/define it in one place and thats `ThemeBackend.js`, I don't see it as big deal. Also then you have the advantage that you basicaly have a directory of every themeable var which you can use as reference for creating a theme.
If you have an scss var you define it twice sometime additionally in `_variables.scss` we could probably resolve the ones in  `_variables.scss` so we only have the ones left in `ThemeBackend.js`.
You're right we could generate the var overwrite to save even more space. I'll try some things.